src/protocol/context/malicious.rs

```diff
@@ -470,6 +473,58 @@ where
     }
 }
 
+enum Upgrade2DVectors {
+    V(usize),
+}
+impl crate::protocol::Substep for Upgrade2DVectors {}
+
+impl AsRef<str> for Upgrade2DVectors {
+    fn as_ref(&self) -> &str {
+        const COLUMN: [&str; 64] = repeat64str!["upgrade_2d"];
+
+        match self {
+            Self::V(i) => COLUMN[*i],
```

Comment:
This will panic at runtime if the input dimension exceeds 64. It looks like many if not all of our existing steps that are implemented this way have the same problem.

For now I guess that's okay since we know our input size is limited? But I worry that the more places this ripples to in the code, the more likely somebody overlooks this restriction and gets a confusing panic.

Maybe it would be better to require that all steps of this kind be constructed with `TryFrom` so that it's clear the construction may fail? Or at least panic on construction of the step with an out of range value, rather than later.

Reply:
Agree on this. This is an assumption we are making currently in the code that steps wont be more than 64

Reply:
I am OK with a panic. I think we could add a "panics" documentation to these things - but as the error would present itself immediately to the developer (and should never crop up in unexpected ways at runtime) I'd rather avoid the annoyance of the codebase having a bunch of casting littered about.